rules/do-you-securely-implement-and-handle-jwt-tokens/rule.md

```diff
@@ -0,0 +1,74 @@
+**Video**: JWTs eveyrthing you need to know (5mins)
+
+<!--endintro-->
+
+### Why JWTs Are Popular
+
+JWTs are favored for their:
+* **Compactness:** Making them easily transmissible via URL, POST parameters, or inside HTTP headers.
+* **Self-contained nature:** Carrying all necessary information, reducing the need for database lookups to authenticate requests.
+* **Versatility:** Being used in various scenarios, from Single Sign-On (SSO) to cross-service authentication in microservices architectures.
+
+### How JWT token are used
+
+JWTs token are used to implement authorised access for resouces (APIs) in an application, here is how a typical flow looks like:
+
+* **User Authentication:** A user logs in to an application, and the server creates and issues a signed JWT token upon successful authentication.
+* **Token Storage:** The client application receives the JWT and stores it securely, typically in local storage or a secure cookie.
+* **Authorised access:** The client sends the JWT with each subsequent request to the server by putting it in the header of the request as a bearer token. The server when processing the request verifies the token's signature and the claims present, for granting access to the requested resource if the token is valid.
+* **Token Expiry:** The server can set an expiry time for the token, after which the client must re-authenticate to obtain a new token. In a typical case a refresh token is also issued with the access token which can be used to obtain a new access token without the user having to re-authenticate.
+
+![Figure: How JWT tokens are used](jwt-how-its-used.png)
+
+### Token Signing Algorithms: Symmetric vs. Asymmetric
+
+When it comes to securing JWTs, the choice of signing algorithm plays a critical role. Here's a breakdown to help you choose:
+
+#### Symmetric Algorithms (e.g., HMAC, HS256)
+
+* **How it works:** Uses a single secret key for both signing and verification processes.
+* **Pros:** Fast computation and straightforward implementation.
+* **Cons:** The secret key must be shared between the issuer and the verifier, posing a risk if the key is exposed.
+* **Best used when:** The issuer and verifier of the JWT are the same entity, or when secure key sharing is feasible.
+
+#### Asymmetric Algorithms (e.g., RSA, ECDSA)
+
+* **How it works:** Utilizes a public/private key pair; the JWT is signed with the private key and verified with the public key.
+* **Pros:** Increased security as the private key never needs to be shared; the public key can be openly distributed for token verification.
+* **Cons:** More computationally intensive, leading to slower performance.
+* **Best used when:** Multiple parties need to verify the token without sharing a secret, such as in distributed systems.
+
+### Best Practices for JWT Implementation
+
+1. **Keep Payloads Lean and Non-sensitive:** Avoid storing sensitive information in the JWT payload. If you must, ensure it's encrypted.
```

Comment:
I think you need to elaborate on this above. JWT payloads are not encrypted so you shouldn't put sensitive information in them. At all. This is just not really good advice, and even if it were, you would need to explain what you mean (i.e. encrypt the content of the claim, plus that's a whole other thing about how do you do that).